nullaway/src/test/java/com/uber/nullaway/NullAwayTest.java

```diff
@@ -1187,4 +1187,9 @@ public void overrideFailsOnExplicitlyNullableLibraryModelParam() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void testCapturingScopes() {
+    compilationHelper.addSourceFile("CapturingScopes.java").doTest();
```

Comment:
Not that I mind too much, but, do we have a guideline of when to do inline tests versus file-wide tests?

Reply:
No guideline.  Would rather not be too picky about this unless it becomes a big problem.  I think the "right" thing would be a separate unit test for each scenario rather than combining scenarios into external source files.  Here I wanted to write a bunch of tests for this new functionality quickly and didn't feel like doing a bunch of copy-paste across distinct unit tests.